applications/vanilla/modules/class.tagmodule.php

```diff
@@ -103,6 +103,13 @@ public function getData() {
 
         $this->autoContext();
 
+        $this->EventArguments['id'] = $this->ParentID;
+        $this->EventArguments['_TagData'] = &$this->_TagData;
+        $this->fireEvent('getData');
+        if($this->_TagData) {
+            return;
```

Comment:
This method has some pretty specific criteria for its data. It should be an instance of `Gdn_DataSet` with a type of `DATASET_TYPE_ARRAY`. Data populated as the result of the getData event should be validated and possibly converted, as necessary, so we're sure what the module is expecting for its data is actually what it's getting.